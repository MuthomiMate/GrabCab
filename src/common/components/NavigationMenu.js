import React, { Component } from "react";
import { NavigationActions } from "react-navigation";
import {View, ScrollView, Text, StyleSheet} from "react-native";
import { Icon, Avatar } from "react-native-elements";

class NavigationMenu extends Component {
    constructor(props){
        super(props)
    }
    componentDidMount(){

    }
    navigateToScreen = ( route ) => {
        const navigationAction = NavigationActions.navigate({
            routeName: route
        });
    this.props.navigation.dispatch(navigationAction)
    }

    render(){
        return (
            <View style={styles.container}>
                <View style={styles.profile}>
                    <Avatar rounded title="MM" size="large"/>
                    <Text style={styles.name}>Muthomi Mate</Text>
                    <View style={{flexDirection: "row"}}>
                        <Icon name="envelope-o" size={15} type="font-awesome" color="#FFFFFF" iconStyle={{marginRight:10}}/>
                        <Text style={{color: "#FFFFFF"}}>muthomimate@gmail.com</Text>
                    </View>
                </View>
                <ScrollView contentContainerStyle={styles.scrollView}>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="user" size={15} type="entypo" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Profile Setttings</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="car-side" size={15} type="material-community" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>My Rides</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="notifications" size={15} type="material-icons" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Notifications</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="payment" size={15} type="material-icons" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Payments</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="cut" size={15} type="font-awesome" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Cuopons</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="language" size={15} type="font-awesome" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>English</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="headset-mic" size={15} type="material-icons" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Support and FAQ</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="info" size={15} type="entypo" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>About us</Text>
                        </View>
                    </View>
                    <View style={styles.menuItem}>
                        <View style={styles.icon}>
                            <Icon name="sign-out" size={15} type="font-awesome" color="#FFFFFF"/>
                        </View>
                        <View>
                            <Text style={{color: "#FFFFFF"}}>Sign Out</Text>
                        </View>
                    </View>
                </ScrollView>
            </View>
        )
    }
}
const styles = StyleSheet.create({
    container: {
        backgroundColor: "#111A1C",
        height: "100%",
    },
    menuItem: {
        flex: 1,
        flexDirection: "row",
        margin: 10,
        width:"100%",
    },
    icon: {
        marginRight: 10,
        height: 20,
        width: 20,
        borderRadius: 50,
        backgroundColor: "#555B5E",
        justifyContent: "center",
    },
    profile: {
        top: "8%",
        justifyContent: "center",
        alignItems: "center",
        textAlign: "center"
    },
    name: {
        color: "#FFFFFF",
        textTransform: "uppercase"
    },
    scrollView: {
        top: "15%"
    },
    iconView: {
        backgroundColor: "#555B5E"
    }

})
export default NavigationMenu;