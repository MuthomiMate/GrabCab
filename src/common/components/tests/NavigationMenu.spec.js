import React from "react";
import { shallow } from "enzyme";
import NavigationMenu from "../NavigationMenu";

describe("<NavigationMenu />", () => {

    it("renders without crashing", () => {
        const wrapper = shallow(<NavigationMenu/>);
        expect(wrapper).toMatchSnapshot();
    })
})