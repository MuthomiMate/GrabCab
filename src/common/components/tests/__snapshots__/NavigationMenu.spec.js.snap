// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationMenu /> renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#111A1C",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "textAlign": "center",
        "top": "8%",
      }
    }
  >
    <Themed.Avatar
      rounded={true}
      size="large"
      title="MM"
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textTransform": "uppercase",
        }
      }
    >
      Muthomi Mate
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#FFFFFF"
        iconStyle={
          Object {
            "marginRight": 10,
          }
        }
        name="envelope-o"
        size={15}
        type="font-awesome"
      />
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        muthomimate@gmail.com
      </Text>
    </View>
  </View>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "top": "15%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="user"
          size={15}
          type="entypo"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Profile Setttings
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="car-side"
          size={15}
          type="material-community"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          My Rides
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="notifications"
          size={15}
          type="material-icons"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Notifications
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="payment"
          size={15}
          type="material-icons"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Payments
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="cut"
          size={15}
          type="font-awesome"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Cuopons
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="language"
          size={15}
          type="font-awesome"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          English
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="headset-mic"
          size={15}
          type="material-icons"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Support and FAQ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="info"
          size={15}
          type="entypo"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          About us
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#555B5E",
            "borderRadius": 50,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 20,
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="sign-out"
          size={15}
          type="font-awesome"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Sign Out
        </Text>
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
